docs/content/docs/core-concept/transform.md

```diff
@@ -28,5 +28,245 @@ under the License.
+
 # Example
-This feature will support soon.
\ No newline at end of file
+## Add computed columns
+If add two computed columns to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, UPPER(product_name) as product_name, localtimestamp as new_timestamp
+    description: project fields from source table
+```
+
+## Reference metadata columns
+If add a metadata columns to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, __namespace_name__ || '.' || __schema_name__ || '.' || __table_name__ identifier_name
+    description: project fields from source table
+```
+
+## Use star instead of all fields
+If use star instead of all fields in the table `web_order` and  `app_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: \*, UPPER(product_name) as product_name
+    description: project fields from source table
+  - source-table: mydb.app_order
+    projection: UPPER(product_name) as product_name, *
+    description: project fields from source table
+```
+Tips: When the star is at the beginning of the rule, a backslash needs to be added for escape.
+
+## Add filter rule
+Use reference columns when adding filtering rules to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    filter: id > 10 AND order_id > 100
+    description: project fields from source table
+```
+
+Use computed columns when adding filtering rules to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, UPPER(province) as new_province 
+    filter: new_province = 'SHANGHAI'
+    description: project fields from source table
+```
+
+## Reassign primary key
+If reassign primary key to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id
+    primary-keys: order_id
+    description: project fields from source table
+```
+
+If reassign composite primary key to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, UPPER(product_name) as product_name
+    primary-keys: order_id,product_name
+    description: project fields from source table
+```
+
+## Reassign partition key
+If reassign partition key to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
```

Comment:
We can reassign the partition key in transform rules. For example, given a table `web_order` in the database `mydb`, we may define a transform rule as follows: